jobserv/api/build.py

```diff
@@ -178,3 +178,16 @@ def promoted_build_get(proj, name):
         )
     )
     return jsendify({"build": _promoted_as_json(Storage(), b)})
+
+
+@blueprint.route("/external-builds/", methods=("POST",))
+def external_build_create(proj):
+    permissions.assert_can_build(proj)
+    p = Project.query.filter(Project.name == proj).first_or_404()
+    d = request.get_json() or {}
```

Comment:
maybe add a check fro trigger-name?
```
if not d.get("trigger-name"):
    raise ApiError(400, 'Missing required parameter: "trigger-name"')
```

Reply:
I meant to make it optional. Do you think it should be mandatory?